### DISCUSSION TITLE
How do I avoid focus being reset with form inside popover?

### DISCUSSION THREAD

**Discussion by cesalberca** [mantinedev/mantine#1518] [Q&A] [answered]
I have this form inside a Popover and when I submit it or close it, the focus is being reset. I would like for the focus to return to the element triggering the Popover. Is this possible? I would like to do something like the example shown here: https://mantine.dev/hooks/use-focus-return/.

I've create the following repro: https://codesandbox.io/s/snowy-tdd-21otff?file=/src/App.tsx.

**Comment by rtivital** [comment, marked as answer]
You can use use-focus-return hook when popover closes, Popover focus management will be improved in 5.0, until that you need to manage that on your side

**Reply by cesalberca** [reply]
That works! Here is the updated Codesandbox for anyone looking for the solution: https://codesandbox.io/s/eager-maxwell-uy4ue5?file=/src/App.tsx
